quay_setup.sh

```diff
@@ -1,18 +1,56 @@
 # oc create -f https://raw.githubusercontent.com/minio/minio-operator/master/minio-operator.yaml
+
+check(){
+  eval VAR1="$1"
+  eval VAR2="$2"
+  while true; do
+    eval $VAR1
+    if [[ $? == $VAR2 ]]
+    then
+      break
+    fi
+    sleep 1
+  done
+}
+
+#Create a Project
+oc new-project quay-enterprise
 # install quay operator
 oc apply -f quay-op.yaml
-# install cso operator
-oc apply -f cso.yaml
 # add pull secrete for quay images
 oc create secret generic redhat-pull-secret --from-file=".dockerconfigjson=docker_quay.json" --type='kubernetes.io/dockerconfigjson'
+#Quay-operator takes some time to come(especially on crc) up so need to wait before bringing up quayecosystem
+
+#code to wait for quay-operator to come up
+req="1"
+com="oc get pods -o json | jq -r '.items[0].status.phase' | grep -v Running"
+check "\${com}" "\${req}"
```

Comment:
Rename check to exec_and_wait. Also it should wait for the given command to return 0, so that you could avoid `req` arg.
```suggestion
exec_and_wait "oc get pods -o json | jq -r '.items[0].status.phase' | grep Running".
```

Reply:
Cannot do that as ```oc get pods -o json | jq -r '.items[0].status.phase' | grep -v Running``` has exir code 1 when successfull.

Reply:
Did you look at my recommended command? use `grep Running` instead of `grep -v Running`

Reply:
Okay.I missed that.Sorry for that.Let me commit the changes.